shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -530,8 +533,10 @@ private void copyToLocal(AlluxioURI srcPath, AlluxioURI dstPath) throws AlluxioE
         try {
           File subDstFile = new File(dstFile.getAbsolutePath(), status.getName());
           copyToLocal(
-              new AlluxioURI(srcPath.getScheme(), srcPath.getAuthority(), status.getPath()),
-              new AlluxioURI(dstPath.getScheme(), dstPath.getAuthority(), subDstFile.getPath()));
+              new AlluxioURI(srcPath.getScheme(), srcPath.hasAuthority()
+                  ? srcPath.getAuthority().toString() : null, status.getPath()),
+              new AlluxioURI(dstPath.getScheme(), dstPath.hasAuthority()
```

Comment:
It's kind of painful how we need to check `hasAuthority()` in all of these places that we construct an `AlluxioURI`. Could we do this handling inside of the `AlluxioURI` constructor instead?

Reply:
I first though creating constructors in Alluxio URI like
```
  public AlluxioURI(String scheme, String authority, String path) {
    mUri = URI.Factory.create(scheme, authority, path, null);
  }

  public AlluxioURI(String scheme, Authority authority, String path) {
    mUri = URI.Factory.create(scheme, authority == null ? null : authority.toString(), path, null);
  }
```
to support both String authority and Authority object, but when it comes to 
```
AlluxioURI uri = new AlluxioURI(null, null, path);
```
It will not be able to resolve the constructor.
If I only expose Authority type constructor, lots of places need to change codes and it may not be that convenience. 

I will consider adding NoAuthority to Authority and see if it works.

Reply:
Great suggestion! 
Changed to use 
```
AlluxioURI(String scheme, Authority authority, String path)
```

and changed From `Authority.Factory.create()` to `Authority.fromString()`.
Java 8 is great!! 

Thanks!